### PR TITLE
feat(windows): add native process host

### LINKED ISSUE #1351: Port the native process host to Windows with fail-closed capabilities
## Problem

Astrid's native process host and process-related WIT implementation contain
Unix-only execution, path, environment, signal, and local-bind assumptions.
Blindly enabling them on Windows would either fail at runtime or silently widen
capabilities.

## Scope

- Introduce a platform process-host backend that preserves the existing WIT and
  public wire shape.
- Implement Windows command-line construction without shell interpolation,
  deterministic environment inheritance/clearing, working-directory checks,
  stdio plumbing, exit status, cancellation, and process-tree termination.
- Make bind-local/network and native MCP behavior explicit per platform.
- Keep unsupported Windows operations denied and audited until their security
  backend exists; never emulate a denied capability with a broader primitive.
- Preserve current Unix behavior.
- Add x86_64 and ARM64 Windows-native process-host tests and capability-denial
  audit tests.

Restricted tokens, Job Objects, filesystem isolation, copy-on-write workspaces,
product integration, signing, and packaging are separate work.

## Acceptance criteria

- No command passes through `cmd.exe`, PowerShell, or another shell unless the
  capsule explicitly invokes that executable under an allowed process policy.
- Argument boundaries, environment entries, working directories, stdio, exit
  codes, cancellation, and descendant cleanup are deterministic.
- Unsupported bind-local/native-MCP operations fail closed with an audit event.
- Unix process-host behavior and the public WIT/API remain unchanged.
- Native Windows x86_64 and ARM64 tests cover quoting edge cases, environment,
  stdio, cancellation, child trees, denied capabilities, and audit records.

### PR DESCRIPTION
**Pull request by joshuajbouw**
## Linked Issue

Closes #1351

## Summary

Adds a native Windows process-host backend without changing Astrid's public WIT
or caller-facing MCP APIs. Windows process execution now preserves argument and
environment boundaries, assigns process trees to a Job Object before execution
begins, and fails closed with typed audit evidence when an OS sandbox or another
capability is unavailable.

## Changes

- construct Windows command lines directly without shell interpolation and
  preserve Unix environment semantics unchanged
- create Windows process trees suspended, assign them to a kill-on-close Job
  Object, then resume execution
- implement deterministic working-directory, environment, stdio, wait,
  cancellation, background, and persistent-process behavior
- preserve the public ProcessTracker API while tracking Windows process trees
  internally
- use executable-only, secret-free process audit descriptors and signed denial
  records
- append the new process-signal audit action so every existing public
  `AuditAction` discriminant remains unchanged
- reuse the workspace-owned `windows-sys` version after rebasing onto the
  merged Windows filesystem boundary
- classify only required sandbox unavailability as `AuthorizationProof::Denied`
  through the real configured and dynamic secure MCP paths, while keeping
  configuration, missing-binary, spawn, and protocol failures as ordinary
  `System`/`Failure` audit outcomes
- warn when explicit managed-process tree cleanup fails while retaining the
  root kill-on-drop and Job Object kill-on-close backstops
- keep sandbox-path UTF-8 rejection tests active on both Unix and Windows by
  constructing platform-native invalid text paths
- explicitly deny unsupported Windows capability paths rather than widening
  them
- add native x64 and ARM64 workflow coverage and regression tests for quoting,
  environment behavior, stdio, process-tree termination, persistent processes,
  and denied capabilities
- keep the dedicated native Clippy gate scoped to its six explicit process-host
  packages while linting all of their targets and features

## Verification

- `cargo +1.95.0 fmt --all -- --check`
- `git diff --check`
- `cargo metadata --locked --offline --no-deps`
- `actionlint .github/workflows/windows-process-host.yml`
- `cargo metadata --locked --offline`
- `cargo +1.95.0 test --locked --offline -p astrid-mcp --no-run`
- `cargo test -p astrid-workspace` (67 tests)
- `cargo test -p astrid-mcp` (125 tests and 2 doctests)
- `cargo test -p astrid-capsule` (581 tests; Windows integration target
  compiles and is exercised natively in CI)
- `cargo clippy -p astrid-capsule -p astrid-workspace -p astrid-mcp
  --all-targets --all-features -- -D warnings`
- dedicated process-host Clippy command with `--no-deps` passes locally; native
  x64 and ARM64 reruns exercise the same Windows-only targets
- targeted sandbox error-classification and non-UTF-8 path regressions
- x64 and ARM64 Windows cross-target compile and Clippy checks performed during
  implementation
- independent adversarial review of process lifecycle, audit classification,
  destructor cleanup diagnostics, environment semantics, public API
  compatibility, and Windows workflow coverage

The native x64 and ARM64 workflow is the merge gate for Windows-only runtime
behavior.

## Checklist

- [x] Linked to an issue
- [x] CHANGELOG.md updated under `[Unreleased]`

### CODE REVIEW

File: crates/astrid-mcp/src/server.rs [hunk 1]
```diff
@@ -568,13 +617,18 @@ impl ServerManager {
...
+            let error = McpError::ServerStartFailed {
+                name: name.to_string(),
+                reason: reason.clone(),
+            };
+            if sandbox_policy == SandboxPolicy::Required {
+                ServerConnectionError::SandboxPolicyDenied { reason, error }
+            } else {
+                ServerConnectionError::Other(error)
+            }
+        })?;
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-mcp/src/server.rs]: Fixed in 58c8a4c. Sandbox unavailability now has `io::ErrorKind::Unsupported`, and only `Required + Unsupported` maps to `SandboxPolicyDenied`; invalid configuration and other construction failures remain `Other` and therefore retain `AuthorizationProof::System`. Added regressions for both classification arms and updated the Windows producer-contract test.

File: crates/astrid-capsule/src/engine/wasm/host/process/managed.rs [hunk 2]
```diff
@@ -42,34 +59,191 @@ pub struct ManagedProcess {
...
-    child.try_wait().ok().flatten().and_then(|s| s.code())
 }
 
 impl Drop for ManagedProcess {
     fn drop(&mut self) {
         if let Some(mut child) = self.child.take() {
-            kill_and_reap(&mut child);
+            let _ = kill_and_reap(&mut child, &self.tree);
         }
     }
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply crates/astrid-capsule/src/engine/wasm/host/process/managed.rs:255]: Fixed in 58c8a4c. `ManagedProcess::drop` now warns with the PID and typed termination error when explicit tree cleanup fails. Drop remains non-panicking and best-effort; Windows still retains the root `kill_on_drop` and Job Object kill-on-close backstops.
